### PR TITLE
Update DG & PPP

### PR DESCRIPTION
**Pull request by Cyn7hius**
- Update DG to include design considerations for help and stats windows
- Added UML diagrams for help and stats windows
- Added manual testing for help window, stats window, and saving data
- Update PPP

### CODE REVIEW

File: docs/diagrams/StatsWindowSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/StatsWindowSequenceDiagram.puml:13]: Is the activation bar missing? Or is it intended to be left out?

File: docs/diagrams/StatsWindowSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/StatsWindowSequenceDiagram.puml:21]: Similar to the previous point. Are the activation bars missing? Or are they intended to be left out?

File: docs/diagrams/StatsWindowSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/StatsWindowSequenceDiagram.puml:22]: Clarification: Is `StatsWindow()` supposed to return something to the caller? Since theres no return arrow from `:StatsWindow` object.

File: docs/diagrams/HelpCommandParsingSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/HelpCommandParsingSequenceDiagram.puml:21]: Is the activation bar missing? Or is it intended to be left out?

File: docs/diagrams/HelpCommandParsingSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/HelpCommandParsingSequenceDiagram.puml:26]: Is the activation bar missing? Or is it intended to be left out?

File: docs/diagrams/HelpCommandParsingSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/HelpCommandParsingSequenceDiagram.puml:13]: Clarification: Is `executeCommand()` supposed to return something to the caller? Since theres no return arrow from `:MainWindow` object.

File: docs/diagrams/HelpCommandParsingSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/HelpCommandParsingSequenceDiagram.puml:30]: Clarification: Is `show()` supposed to return something to the caller? Since theres no return arrow from `:HelpWindow` object

File: docs/diagrams/HelpCommandParsingSequenceDiagram.puml
Comment (Py0000) [inline docs/diagrams/HelpCommandParsingSequenceDiagram.puml:3]: Is the background color for the different components missing?

File: docs/DeveloperGuide.md
Comment (Py0000) [inline docs/DeveloperGuide.md:1627]: I think you might have forgotten to remove this line?

File: docs/DeveloperGuide.md
Comment (Py0000) [inline docs/DeveloperGuide.md:1646]: Similarly, you might have forgotten to remove this line after resolving the merge conflict

File: docs/DeveloperGuide.md
Comment (Py0000) [inline docs/DeveloperGuide.md:1653]: Similarly for this as well, you might have forgotten to remove this line after resolving the merge conflict

Comment (Cyn7hius) [reply docs/DeveloperGuide.md]: Thanks for pointing it out!

**Comment by Cyn7hius** [comment]
Thanks for point out the missing parts for my diagram!

**Review by Py0000 — APPROVED** [review]
LGTM after the update.

Small thing to take note/discuss: Color issue for StatsWindowSequenceDiagram

**Pull request merged**
